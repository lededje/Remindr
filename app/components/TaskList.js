import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  ListView,
  Text,
} from 'react-native';
import Task from './Task';

export default class List extends Component {

  static propTypes = {
    tasks: React.PropTypes.arrayOf(React.PropTypes.shape({
      id: React.PropTypes.number.isRequired,
      title: React.PropTypes.string.isRequired,
      timestamp: React.PropTypes.number.isRequired,
    })).isRequired,
  };

  static defaultProps = {
    tasks: [],
  }

  constructor(props) {
    super(props);

    this.dataSource = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
  }

  renderRow(task) {
    return (
      <Task id={task.id} title={task.title} timestamp={task.timestamp} />
    );
  }

  render() {
    const dataSource = this.dataSource.cloneWithRows(this.props.tasks);

    return (
      <View style={styles.container}>
        {this.props.tasks.length === 0 && (
          <ListView
            ref={(lv) => { this.listView = lv; }}
            dataSource={dataSource}
            renderRow={this.renderRow}
          />
        )}
        {this.props.tasks.length > 0 && (
          <Text>This list is empty</Text>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
