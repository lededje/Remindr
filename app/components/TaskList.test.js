import React from 'react';
import TaskList from './TaskList';
import renderer from 'react-test-renderer';

const taskData = [{
  id: 0,
  title: 'Test task 1',
  timestamp: 1,
}, {
  id: 1,
  title: 'Test task 2',
  timestamp: 2,
}];

it('renders the task list component', () => {
  const component = renderer.create(
    <TaskList tasks={taskData} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
