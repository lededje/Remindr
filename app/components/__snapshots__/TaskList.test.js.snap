exports[`test renders an empty list component with message 1`] = `
<View
  style={
    Object {
      "flex": 1
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 20,
        "textAlign": "center"
      }
    }>
    This list is empty
  </Text>
</View>
`;

exports[`test renders the task list component 1`] = `
<View
  style={
    Object {
      "flex": 1
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 2
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function bound _onContentSizeChange]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function bound _onLayout]}
    onScroll={[Function bound _onScroll]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function renderRow]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#eee",
          "paddingBottom": 20,
          "paddingLeft": 35,
          "paddingRight": 35,
          "paddingTop": 20
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "300",
            "marginBottom": 10
          }
        }>
        Test task 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999",
            "fontSize": 12,
            "fontWeight": "300"
          }
        }>
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#eee",
          "paddingBottom": 20,
          "paddingLeft": 35,
          "paddingRight": 35,
          "paddingTop": 20
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "300",
            "marginBottom": 10
          }
        }>
        Test task 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999",
            "fontSize": 12,
            "fontWeight": "300"
          }
        }>
        2
      </Text>
    </View>
  </ScrollView>
</View>
`;
